crates/title_bar/src/title_bar.rs

```diff
@@ -379,18 +380,23 @@ impl Render for TitleBar {
             .when(
                 self.platform_style == PlatformStyle::Linux
                     && !cx.is_fullscreen()
-                    && cx.should_render_window_controls(),
+                    && matches!(decorations, Decorations::Client { .. }),
                 |title_bar| {
                     title_bar
                         .child(platform_linux::LinuxWindowControls::new(height, close_action))
-                        .on_mouse_down(gpui::MouseButton::Right, move |ev, cx| {
-                            cx.show_window_menu(ev.position)
-                        })
-                        .on_mouse_move(move |ev, cx| {
-                            if ev.dragging() {
```

Comment:
Removing this check makes it impossible to press any buttons on the title bar on Wayland or to double click to maximize. It also changes to a grabbing cursor every time I click on the title bar on XWayland.

Reply:
This is now fixed! :smile: The PR description could tag https://github.com/zed-industries/zed/issues/12639